### PR TITLE
chore(argo-cd): Remove deprecated features

### PR DESCRIPTION
**Pull request by pdrastil**
Removed support to configure / enable deprecated features. This PR is opened to make more clear what is deprecated after argocmd-cm-params CM is merged.

Edit: I'm also fine to use this as start of 5.x release that will refactor / move things around in configs and convert PR to Draft that will be extended.

Checklist:

* [x] I have bumped the chart version according to [versioning](https://github.com/argoproj/argo-helm/blob/main/CONTRIBUTING.md#versioning)
* [x] I have updated the documentation according to [documentation](https://github.com/argoproj/argo-helm/blob/main/CONTRIBUTING.md#documentation)
* [x] I have updated the chart changelog with all the changes that come with this pull request according to [changelog](https://github.com/argoproj/argo-helm/blob/main/CONTRIBUTING.md#changelog).
* [x] Any new values are backwards compatible and/or have sensible default.
* [x] I have signed off all my commits as required by [DCO](https://github.com/argoproj/argoproj/blob/master/community/CONTRIBUTING.md).
* [x] My build is green ([troubleshooting builds](https://argo-cd.readthedocs.io/en/stable/developer-guide/ci/)).

Changes are automatically published when merged to `main`. They are not published on branches.

### CODE REVIEW

File: charts/argo-cd/README.md [hunk 1]
```diff
@@ -252,7 +256,6 @@ NAME: my-release
 | configs.knownHostsAnnotations | object | `{}` | Known Hosts configmap annotations |
 | configs.repositories | object | `{}` | Repositories list to be used by applications |
 | configs.repositoriesAnnotations | object | `{}` | Annotations to be added to `configs.repositories` Secret |
-| configs.repositoryCredentials | object | `{}` | *DEPRECATED:* Instead, use `configs.credentialTemplates` and/or `configs.repositories` |
```
Comment (mkilchhofer) [inline charts/argo-cd/README.md:255]: I think these features are deprecated in argo-cd itself but still are supported.
But I am fine with removing them.

Comment (pdrastil) [reply charts/argo-cd/README.md:255]: These are still supported but users were advised to migrate 6 months ago. Even upstream documentation no longer have samples for these.

Comment (mbevc1) [reply charts/argo-cd/README.md:255]: Same as @mkilchhofer I'm happy to remove this, but if it's still a supported, perhaps we could keep it a bit longer for compatibility? Happy to hear other thought on this one.

Comment (jmeridth) [reply charts/argo-cd/README.md:255]: If upstream documentation no longer have samples and advised migration 6 months ago, I think enough time has passed, correct?  I'm good with removal.

**Review by jmeridth — COMMENTED** [review]
LGTM, just gotta fix Chart.yaml conflict.  Will approve after.

**Comment by mkilchhofer** [comment]
@yu-croco maybe we can merge the 5.0.0 pr #1357 with this before release?

**Comment by pdrastil** [comment]
@mkilchhofer The reason I've put these separately without 5.0.0 PR is because of default values and that users should be migrated. Otherwise we can run this in 5.0.0 with all the refactoring stuff for argo-cmd-config.

**Comment by yu-croco** [comment]
@mkilchhofer @pdrastil 
I'll take your lead on this one, please decide whichever makes it go smoothly. 🙋

**Comment by mkilchhofer** [comment]
> @mkilchhofer The reason I've put these separately without 5.0.0 PR is because of default values and that users should be migrated. Otherwise we can run this in 5.0.0 with all the refactoring stuff for argo-cmd-config.

I created a proposal which merges the work of you and @yu-croco:
- #1417

Maybe you did not get what I meant above :-)

**Comment by mkilchhofer** [comment]
Implemented in PR https://github.com/argoproj/argo-helm/pull/1417

**Pull request closed without merging**